packages/http/libraries/express/src/adapter/InversifyExpressHttpAdapter.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  HttpAdapterOptions,
+  HttpStatusCode,
+  InversifyHttpAdapter,
+  RequestHandler,
+  RouterParams,
+} from '@inversifyjs/http-core';
+import express, {
+  Application,
+  NextFunction,
+  Request,
+  Response,
+  Router,
+} from 'express';
+import { Container } from 'inversify';
+
+export class InversifyExpressHttpAdapter extends InversifyHttpAdapter<
+  Request,
+  Response,
+  NextFunction
+> {
+  readonly #app: Application;
+
+  constructor(
+    container: Container,
+    httpAdapterOptions?: HttpAdapterOptions,
+    customApp?: Application,
+  ) {
+    super(container, httpAdapterOptions);
+    this.#app = customApp ?? express();
+  }
+
+  public build(): Application {
+    this.#app.use(express.json());
+
+    this._buildServer();
+
+    return this.#app;
+  }
+
+  protected override _buildRouter(
+    path: string,
+    routerParams: RouterParams<Request, Response, NextFunction>[],
+    middlewareList:
+      | RequestHandler<Request, Response, NextFunction>[]
+      | undefined,
+    guardList: RequestHandler<Request, Response, NextFunction>[] | undefined,
+  ): void {
+    const router: Router = Router();
+
+    const orderedMiddlewareList:
+      | RequestHandler<Request, Response, NextFunction>[]
+      | undefined = this.#orderMiddlewares(guardList, middlewareList);
+
+    if (orderedMiddlewareList !== undefined) {
+      router.use(orderedMiddlewareList);
+    }
+
+    for (const route of routerParams) {
+      const orderedMiddlewareList:
+        | RequestHandler<Request, Response, NextFunction>[]
+        | undefined = this.#orderMiddlewares(
+        route.guardList,
+        route.middlewareList,
+      );
+
+      if (orderedMiddlewareList !== undefined) {
+        router[route.requestMethodType](
+          route.path,
+          ...orderedMiddlewareList,
+          route.handler,
+        );
+      } else {
+        router[route.requestMethodType](route.path, route.handler);
```

Comment:
Express types suck here, we need do cast `router[route.requestMethodType]` to accept async request param handlers